### PR TITLE
[mlir][TilingInterface] NFC code changes separated out from introduction of `scf::tileUsingSCFForallop`.

### PR DESCRIPTION
**Pull request by MaheshRavishankar**
This patch contains NFC changes that are precursor to the introduction of `scf::tileUsingSCFForallOp` method introduced in https://github.com/llvm/llvm-project/pull/67083.

### CODE REVIEW

**Review by nicolasvasilache — APPROVED** [review]

File: mlir/lib/Dialect/SCF/Transforms/TileUsingInterface.cpp [hunk 1]
```diff
@@ -55,6 +55,30 @@ fillInterchangeVector(ArrayRef<int64_t> interchangeVector,
...
+}
+template <typename SrcOpTy>
+static SmallVector<Operation *>
+getAsOperations(const SmallVector<SrcOpTy> &ops) {
+  return getAsOperations(ArrayRef<SrcOpTy>(ops));
+}
+
+/// Convert a list of `Operation *` to a list of `DstOpTy`
+template <typename DstOpTy>
+static SmallVector<DstOpTy> castToTypedOperations(ArrayRef<Operation *> ops) {
```
Comment (chelini) [inline mlir/lib/Dialect/SCF/Transforms/TileUsingInterface.cpp:72]: Do we really need this helper function? Seems to be used only once.

File: mlir/lib/Dialect/SCF/Transforms/TileUsingInterface.cpp [hunk 2]
```diff
@@ -55,6 +55,30 @@ fillInterchangeVector(ArrayRef<int64_t> interchangeVector,
...
+  return llvm::to_vector(
+      llvm::map_range(ops, [](auto op) -> Operation * { return op; }));
+}
+template <typename SrcOpTy>
+static SmallVector<Operation *>
+getAsOperations(const SmallVector<SrcOpTy> &ops) {
+  return getAsOperations(ArrayRef<SrcOpTy>(ops));
+}
+
+/// Convert a list of `Operation *` to a list of `DstOpTy`
```
Comment (chelini) [inline mlir/lib/Dialect/SCF/Transforms/TileUsingInterface.cpp:70]: nit: "." at the end of the sentence.

File: mlir/test/lib/Interfaces/TilingInterface/TestTilingInterface.cpp [hunk 3]
```diff
@@ -37,51 +37,51 @@ using namespace mlir;
 namespace {
 
 /// Marker used as attribute name in generated Linalg rewriting transformations.
-const StringLiteral kLinalgTransformMarker = "__internal_linalg_transform__";
+const StringLiteral kTransformMarker = "__internal_linalg_transform__";
```
Comment (chelini) [inline mlir/test/lib/Interfaces/TilingInterface/TestTilingInterface.cpp:40]: drop "linalg" from the merker?

File: mlir/lib/Dialect/SCF/Transforms/TileUsingInterface.cpp (hunk 1 above)
Comment (MaheshRavishankar) [reply mlir/lib/Dialect/SCF/Transforms/TileUsingInterface.cpp:72]: These are the mirror of the ones above. Makes it easier to follow the code later on. I am going to leave it there for now?

**Review by chelini — APPROVED** [review]

**Pull request merged**